tests/lib/fakeinstaller/main.go

```diff
@@ -36,9 +40,78 @@ import (
 	"github.com/snapcore/snapd/osutil"
 	"github.com/snapcore/snapd/osutil/disks"
 	"github.com/snapcore/snapd/osutil/mkfs"
-	"github.com/snapcore/snapd/overlord/state"
 )
 
+func waitForDevice() string {
+	for {
+		devices, err := emptyFixedBlockDevices()
+		if err != nil {
+			logger.Noticef("cannot list devices: %v", err)
+		}
+		switch len(devices) {
+		case 0:
+			logger.Noticef("cannot use automatic mode, no empty disk found")
+		case 1:
+			// found exactly one target
+			return devices[0]
+		default:
+			logger.Noticef("cannot use automatic mode, multiple empty disks found: %v", devices)
+		}
+		time.Sleep(5 * time.Second)
+	}
+}
+
+// emptyFixedBlockDevices finds any non-removable physical disk that has
+// no partitions. It will exclude loop devices.
+func emptyFixedBlockDevices() (devices []string, err error) {
+	// eg. /sys/block/sda/removable
+	removable, err := filepath.Glob(filepath.Join(dirs.GlobalRootDir, "/sys/block/*/removable"))
+	if err != nil {
+		return nil, err
+	}
+devicesLoop:
+	for _, removableAttr := range removable {
+		val, err := ioutil.ReadFile(removableAttr)
+		if err != nil || string(val) != "0\n" {
+			// removable, ignore
+			continue
+		}
+		dev := filepath.Base(filepath.Dir(removableAttr))
+		if strings.HasPrefix(dev, "loop") {
+			// is loop device, ignore
+			continue
+		}
+		// let's see if it has partitions
+		pattern := fmt.Sprintf(filepath.Join(dirs.GlobalRootDir, "/sys/block/%s/%s*/partition"), dev, dev)
+		// eg. /sys/block/sda/sda1/partition
+		partitionAttrs, _ := filepath.Glob(pattern)
+		if len(partitionAttrs) != 0 {
+			// has partitions, ignore
+			continue
+		}
+		// check that there was no previous filesystem
+		devNode := fmt.Sprintf("/dev/%s", dev)
+		output, err := exec.Command("lsblk", "--output", "fstype", "--noheadings", devNode).CombinedOutput()
+		if err != nil {
+			return nil, osutil.OutputErr(output, err)
+		}
+		scanner := bufio.NewScanner(strings.NewReader(string(output)))
+		for scanner.Scan() {
+			if scanner.Text() != "" {
+				// found a filesystem, ignore
+				continue devicesLoop
+			}
+		}
+		if err := scanner.Err(); err != nil {
+			return nil, err
+		}
```

Comment:
Can't we just do
```suggestion
		if strings.TrimSpace(string(output)) != "" {
			// found a filesystem, ignore
			continue devicesLoop
		}
```
?

If not, I think I comment would help.

Reply:
I used the scanner because you can get more than one filesystem (for instance from encrypted fs) so you get a few lines, but here we just want to check if there is any fs, so I agree with your approach. Thanks!